src/test/scala/nl.knaw.dans.easy.deposit/docs/DatasetXmlSpec.scala

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (C) 2018 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit.docs
+
+import better.files.File
+import nl.knaw.dans.easy.deposit.TestSupportFixture
+
+import scala.util.{ Failure, Success, Try }
+
+class DatasetXmlSpec extends TestSupportFixture {
+
+  private val prettyPrinter = new scala.xml.PrettyPrinter(1024, 2)
+  private val minimal = DatasetMetadata(
+    """{
+      |  "titles": [""],
+      |  "descriptions": [""],
+      |  "dates": [
+      |    { "scheme": "", "value": "", "qualifier": "dcterms:created" },
+      |    { "scheme": "", "value": "", "qualifier": "dcterms:available" },
+      |  ],
+      |  "creators": [ { "initials": "", "surname": "" } ],
+      |  "accessRights": { "category": "OPEN_ACCESS" },
+      |  "audiences": [ { "scheme": "", "key": "", "value": ""} ]
+      |}""".stripMargin).getOrElse(fail("parsing minimal json failed"))
+
+  "apply" should "produce DDM from  minimal json" in {
+    DatasetXml(minimal) shouldBe a[Success[_]]
+  }
+
+  it should "report a missing title" in {
+    DatasetXml(minimal.copy(titles = None)) should matchPattern {
+      case Failure(e) if e.getMessage == "no content for mandatory dcterms:title" =>
```

Comment:
👍 for this `case ... if ... =>` syntax
I don't think I've ever seen that before, but that's really nice. Probably also gives a better error message than using `inside` for this.